src/test/java/com/reservation/instructor/domain/InstructorTest.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Test
+    void shouldCreateInstructorWithActiveStatus() {
+        events = instructor.pullEvents();
+
+        assertEquals(instructorId, instructor.getInstructorId());
+        assertEquals(systemUserId, instructor.getSystemUserId());
+        assertTrue(instructor.isSameProfile(instructorProfile));
+        assertTrue(instructor.isActive());
+
+        assertEquals(1, events.size());
+        InstructorEvent event = events.getFirst();
+        assertInstanceOf(InstructorCreated.class, event);
+        InstructorCreated createdEvent = (InstructorCreated) event;
+        assertEquals(instructorId, createdEvent.instructorId());
+        assertEquals(systemUserId, createdEvent.systemUserId());
+        assertNotNull(createdEvent.createdAt());
+    }
+
+    @Test
+    void shouldUpdateProfileWhenActive() {
+        Name newName = new Name("Jane Smith");
+        Specialty newSpecialty = new Specialty("Pilates");
+        InstructorProfile newProfile = new InstructorProfile(newName, Set.of(newSpecialty), new Bio("New bio"));
+
+        instructor.updateProfile(newProfile);
+        events = instructor.pullEvents();
+
+        assertTrue(instructor.isSameProfile(newProfile));
+        assertEquals(2, events.size());
+        InstructorEvent event = events.get(1);
+        assertInstanceOf(InstructorUpdated.class, event);
+        InstructorUpdated updatedEvent = (InstructorUpdated) event;
+        assertEquals(instructorId, updatedEvent.instructorId());
+        assertEquals(systemUserId, updatedEvent.systemUserId());
+        assertNotNull(updatedEvent.updatedAt());
+    }
+
+    @Test
+    void shouldNotAddUpdateEventIfProfileIsSame() {
+        instructor.updateProfile(instructorProfile);
+        events = instructor.pullEvents();
+
+        assertEquals(1, events.size());
+    }
+
+    @Test
+    void shouldThrowExceptionWhenUpdatingProfileWhileInactive() {
+        Name newName = new Name("Jane Smith");
+        Specialty newSpecialty = new Specialty("Pilates");
+        InstructorProfile newProfile = new InstructorProfile(newName, Set.of(newSpecialty), new Bio("New bio"));
+
+        instructor.deactivate();
+
+        assertThrows(ProfileCannotBeUpdatedWhenInactiveException.class, () ->
+                instructor.updateProfile(newProfile)
+        );
+    }
+
+    @Test
+    void shouldActivateInstructorWhenInactive() {
+        instructor.deactivate();
+        instructor.activate();
+        events = instructor.pullEvents();
+
+        assertTrue(instructor.isActive());
+        assertEquals(3, events.size());
+        InstructorEvent event = events.get(2);
+        assertInstanceOf(InstructorActivated.class, event);
+        InstructorActivated activeEvent = (InstructorActivated) event;
+        assertEquals(instructorId, activeEvent.instructorId());
+        assertEquals(systemUserId, activeEvent.systemUserId());
+        assertNotNull(activeEvent.activationAt());
+    }
+
```

Comment:
Sugestia drobnej poprawy testu pod kątem czytelności i podejścia DDD ->

Obecnie test zakłada, że `InstuctorActivated` będzie trzecim zdarzeniem `events.get(2)` co czyni go zależnym od historii innych operacji na agregacie, w testach domenowych lepiej kierować się pytaniem "Czy zdarzyło się x?" a nie czy X było pod indeksem 2. Chodzi o to aby uniezależnić test od sekwencji zdarzeń i uodpornić go na przyszłe zmiany.


```suggestion
  @Test
  void shouldActivateInstructorWhenInactive() {
      instructor.deactivate();
      instructor.activate();
      events = instructor.pullEvents();
  
      assertTrue(instructor.isActive());
  
      InstructorActivated activatedEvent = events.stream()
          .filter(InstructorActivated.class::isInstance)
          .map(InstructorActivated.class::cast)
          .findFirst()
          .orElseThrow(() -> new AssertionError("Expected InstructorActivated event"));
  
      assertAll(
          () -> assertEquals(instructorId, activatedEvent.instructorId()),
          () -> assertEquals(systemUserId, activatedEvent.systemUserId()),
          () -> assertNotNull(activatedEvent.activationAt())
      );
  }

```